### PR TITLE
Awesome es6

### PR DESCRIPTION
**Pull request by mengiefen**
In this project, the following features have been included

-  add/remove books from a list.
-  the feature to dynamically modify the DOM and add basic events.
-  navigation across the page generally implemented using SPA approach.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,12 +1,74 @@
 <!DOCTYPE html>
 <html lang="en">
+
 <head>
   <meta charset="UTF-8">
   <meta http-equiv="X-UA-Compatible" content="IE=edge">
   <meta name="viewport" content="width=device-width, initial-scale=1.0">
   <title>Awesome Books</title>
+  <!-- <base href="/awesome-books-es6/">  -->
```
Comment (er-contreras) [inline index.html]: - [x] Avoid letting commented code behind to keep the code much cleaner. 👍🏼

File: index.html [hunk 2]
```diff
@@ -1,12 +1,74 @@
...
+  <link rel="stylesheet" href="src/css/index.css">
 </head>
+
 <body>
-  
+
+  <!-- Page Header -->
```
Comment (er-contreras) [inline index.html]: - [x] Avoid letting commented code behind to keep the code much cleaner. 👍🏼

File: index.html [hunk 3]
```diff
@@ -1,12 +1,74 @@
...
+    <h2 class="logo">Awesome Books</h2>
+    <ul class="navbar">
+      <li class="nav-item"><a class="nav-link" href="#book-list">List</a></li>
+      <li class="nav-item"><a class="nav-link" href="#add-new-book">Add new</a>
+      </li>
+      <li class="nav-item"><a class="nav-link" href="#contact">Contact</a></li>
+    </ul>
+  </header>
+
+  <!-- Shows current date -->
```
Comment (er-contreras) [inline index.html]: - [x] Avoid letting commented code behind to keep the code much cleaner. 👍🏼

File: index.html [hunk 4]
```diff
@@ -1,12 +1,74 @@
...
+  <div id="current-date"></div>
+  <section id="book-list">
+    <h1 class="heading1">All Awesome Books</h1>
+    <div id="book-shelf">
+    
+    </div>
+  </section>
+
+  <!-- Add New Page -->
```
Comment (er-contreras) [inline index.html]: - [x] Avoid letting commented code behind to keep the code much cleaner. 👍🏼

File: index.html [hunk 5]
```diff
@@ -1,12 +1,74 @@
...
+          <input type="text" id="author" name="book-author"
+            placeholder="Author">
+        </li>
+      </ul>
+      <span class="error">Please enter all require fields.</span>
+      <button id="btn-add" type="button">Add new book</button>
+    </form>
+  </section>
+
+  <!-- Contact Section -->
```
Comment (er-contreras) [inline index.html]: - [x] Avoid letting commented code behind to keep the code much cleaner. 👍🏼

File: index.html [hunk 6]
```diff
@@ -1,12 +1,74 @@
...
+      Do you have any questions or you just want to say 'Hello'?
+    </p>
+    <ul class="specific-info">
+      <li class="this-info">Our e-mail: mail@mail.com</li>
+      <li class="this-info">Our phone number: 0043586534422</li>
+      <li class="this-info">Our address: Streetname 22, 84503 City, Country</li>
+    </ul>
+  </section>
+
+  <!-- Page Footer -->
```
Comment (er-contreras) [inline index.html]: - [x] Avoid letting commented code behind to keep the code much cleaner. 👍🏼

File: index.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
+import formattedDate from './scripts/modules/date.js';
+import { newDate, addButton } from './scripts/modules/elements.js';
+import handleNavigation from './scripts/modules/navigation.js';
+import addSingleBook from './scripts/modules/addBook.js';
+import redirectToList from './scripts/modules/redirect.js';
+import drawAllBooks from './scripts/modules/reRender.js';
+import showError from './scripts/modules/error.js';
+import book from './scripts/modules/createNewBook.js';
+
+// Show current time
```
Comment (er-contreras) [inline index.js]: - [x] Avoid letting commented code behind to keep the code much cleaner. 👍🏼

File: index.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+newDate.innerHTML = formattedDate;
+
+// Renders book at page load
```
Comment (er-contreras) [inline index.js]: - [x] Avoid letting commented code behind to keep the code much cleaner. 👍🏼

File: index.js [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+drawAllBooks();
+
+// Event listener for 'Add new Button'
```
Comment (er-contreras) [inline index.js]: - [x] Avoid letting commented code behind to keep the code much cleaner. 👍🏼

File: index.js [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+addButton.addEventListener('click', () => {
+  const status = addSingleBook();
+  if (status) {
+    redirectToList();
+  } else {
+    showError();
+  }
+});
+
+// Event listener for 'Add new Button', and Nav Items
```
Comment (er-contreras) [inline index.js]: - [x] Avoid letting commented code behind to keep the code much cleaner. 👍🏼

File: scripts/modules/addBook.js [hunk 11]
```diff
@@ -0,0 +1,26 @@
+import displayBook from './displayBook.js';
+import book from './createNewBook.js';
+import {
+  titleInput, authorInput, form, error,
+} from './elements.js';
+
+// Add single book and writes to local storage
```
Comment (er-contreras) [inline scripts/modules/addBook.js]: - [ ] Avoid letting commented code behind to keep the code much cleaner. 👍🏼

File: scripts/modules/book.js [hunk 12]
```diff
@@ -0,0 +1,42 @@
+export default class Book {
+  constructor(title, author, id) {
+    this.id = id;
+    this.title = title;
+    this.author = author;
+  }
+
+  Books = [];
+
+  // Adds a single book and store in Book
```
Comment (er-contreras) [inline scripts/modules/book.js:10]: - [x] Avoid letting commented code behind to keep the code much cleaner. 👍🏼

File: scripts/modules/book.js [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+    });
+    return this.Books;
+  }
+
+  // Reads books from local storage
+  readBooks() {
+    const bookSaved = JSON.parse(localStorage.getItem('books'));
+    if (bookSaved) {
+      this.Books = bookSaved;
+    }
```
Comment (er-contreras) [inline scripts/modules/book.js:25]: - [x] Avoid letting commented code behind to keep the code much cleaner. 👍🏼

**Review by er-contreras — CHANGES_REQUESTED** [review]
Hi @mengiefen, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

almost there

## Required Changes ♻️
- [x] I like the way you used luxon taking it from the node modules but there is an empty module it would be great to just delete scripts/modules/luxon.js

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ShoiraTa — APPROVED** [review]
### **STATUS: APPROVED  ✔️** 
------
Hi @mengiefen ,  

Your project is complete! Everything works as expected! Congratulations!  🎉💯🌟 



Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**